lib/ui/painting/image_encoding.h

```diff
@@ -5,6 +5,10 @@
 #ifndef FLUTTER_LIB_UI_PAINTING_IMAGE_ENCODING_H_
 #define FLUTTER_LIB_UI_PAINTING_IMAGE_ENCODING_H_
 
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "third_party/skia/include/core/SkCanvas.h"
+#include "third_party/skia/include/core/SkImage.h"
+#include "third_party/skia/include/core/SkSurface.h"
```

Comment:
This is where I am a little hesitant to use this approach. We have included more header files than before

Reply:
Just pull `ConvertToRasterUsingResourceContext` into its own header file and include it in `image_encoding.cc` and `image_encoding_unittests.cc`.  Name it something like `image_encoding_impl.h`.  It doesn't need to be publicly visible from `image_encoding.h`

Reply:
Done